src/or/config.c

```diff
@@ -6304,10 +6304,8 @@ parse_dir_authority_line(const char *line, dirinfo_type_t required_type,
     smartlist_del_keeporder(items, 0);
   }
 
-  while (smartlist_len(items)) {
+  while (smartlist_len(items) > 2) {
```

Comment:
We can't reserve **the last two items** for the address and fingerprint, because the authority format has always supported space-separated fingerprints. A space separated fingerprint can occupy 1 to 40 items, but typically occupies 10.

Instead, please parse the flags, and then parse **all remaining items** as the address, followed by a space-separated fingerprint.

Reply:
> We probably should add [fingerprint...] at the end to indicate there could be more than one...

[fingerprint…] means "zero or more fingerprints".

There must be exactly one fingerprint, but it can be space-separated. We could document that Tor allows spaces in the fingerprint in the man page.

> We could decide we've finished parsing flags and reached "addr:port" if the current smartlist item contains a ":" character (unless it's possible for a flag to contain that character, either now or in the future, in which case we're back to not having a good way to know we've reached that field).

The ipv6 flag contains a colon. It looks like "ipv6=[2001:feed::1]:12345"

Let's check for a field with a colon, and without an equals sign.
Just stopping when we have exhausted all known flags makes it hard to add new flags.

> Alternatively, we could decide to ban non-IP "address" fields outright, because of the potential security vulnerability introduced by adding DNS to the mix

As I said in the review and the trac ticket, I think we should ban them in the public network, but allow the, in test networks and when using non-default authorities.